### ISSUE TITLE
Workload deploy fails with failed to allocate for range 0: no IP addresses available in range set

### ISSUE THREAD

**Issue by gioppoluca** [rancher/rancher#14504]
**Rancher versions:**
rancher/server or rancher/rancher: 2.0.5
rancher/agent or rancher/rancher-agent: 2.0.5

**Infrastructure Stack versions:**
healthcheck:
ipsec:
network-services:
scheduler:
kubernetes (if applicable):

**Docker version: (`docker version`,`docker info` preferred)**
Containers: 22
 Running: 13
 Paused: 0
 Stopped: 9
Images: 19
Server Version: 18.03.1-ce
Storage Driver: overlay2
 Backing Filesystem: xfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 773c489c9c1b21a6d78b5c538cd395416ec50f88
runc version: 4fc53a81fb7c994640722ac585fa9ca548971871
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-862.3.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 3.701GiB
Name: tst-kubenode1.pochdp.csi.it
ID: WERW:QGCW:DVIW:2SYR:V2NW:PFJ5:UW6M:PY3Y:IH7K:F3PQ:NSDX:QMPD
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
HTTP Proxy: http://proxy-srv.csi.it:3128
No Proxy: localhost,127.0.0.0/8,tst-kubemaster.pochdp.csi.it,tst-kubenode1.pochdp.csi.it,10.102.81.250,0.0.0.0,10.102.81.249
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
3.10.0-862.3.2.el7.x86_64
**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
VMware
**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node rancher + second host for cluster node
**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
deployment of a simple workload using an image from dockerhub
**Steps to Reproduce:**

**Results:**
I try to deploy a workload using apache/zeppelin image from dockerhub and I get the following errors in the kubelet container

**------ LOGS from kube-controller-manager**
I0711 14:47:11.157271       1 event.go:218] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"myzep", Name:"myzep", UID:"4adba2cb-8519-11e8-9f37-005056021169", APIVersion:"extensions", ResourceVersion:"118180", FieldPath:""}): type: 'Normal' reason: 'ScalingReplicaSet' Scaled up replica set myzep-79945ccb7f to 1
I0711 14:47:11.157488       1 replica_set.go:477] Too few replicas for ReplicaSet myzep/myzep-79945ccb7f, need 1, creating 1
I0711 14:47:11.183490       1 deployment_controller.go:485] Error syncing deployment myzep/myzep: Operation cannot be fulfilled on deployments.extensions "myzep": the object has been modified; please apply your changes to the latest version and try again
I0711 14:47:11.193359       1 event.go:218] Event(v1.ObjectReference{Kind:"ReplicaSet", Namespace:"myzep", Name:"myzep-79945ccb7f", UID:"4adda197-8519-11e8-9f37-005056021169", APIVersion:"extensions", ResourceVersion:"118182", FieldPath:""}): type: 'Normal' reason: 'SuccessfulCreate' Created pod: myzep-79945ccb7f-rz8ld
I0711 14:47:11.205454       1 deployment_controller.go:485] Error syncing deployment myzep/myzep: Operation cannot be fulfilled on deployments.extensions "myzep": the object has been modified; please apply your changes to the latest version and try again

**----- LOGS from kubelet **

E0711 15:11:17.994634   24727 remote_runtime.go:92] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "myzep-79945ccb7f-rz8ld_myzep" network: failed to allocate for range 0: no IP addresses available in range set: 10.42.0.1-10.42.0.254
E0711 15:11:17.994678   24727 kuberuntime_sandbox.go:54] CreatePodSandbox for pod "myzep-79945ccb7f-rz8ld_myzep(4ae2623a-8519-11e8-9f37-005056021169)" failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "myzep-79945ccb7f-rz8ld_myzep" network: failed to allocate for range 0: no IP addresses available in range set: 10.42.0.1-10.42.0.254
E0711 15:11:17.994703   24727 kuberuntime_manager.go:646] createPodSandbox for pod "myzep-79945ccb7f-rz8ld_myzep(4ae2623a-8519-11e8-9f37-005056021169)" failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "myzep-79945ccb7f-rz8ld_myzep" network: failed to allocate for range 0: no IP addresses available in range set: 10.42.0.1-10.42.0.254
E0711 15:11:17.994753   24727 pod_workers.go:186] Error syncing pod 4ae2623a-8519-11e8-9f37-005056021169 ("myzep-79945ccb7f-rz8ld_myzep(4ae2623a-8519-11e8-9f37-005056021169)"), skipping: failed to "CreatePodSandbox" for "myzep-79945ccb7f-rz8ld_myzep(4ae2623a-8519-11e8-9f37-005056021169)" with CreatePodSandboxError: "CreatePodSandbox for pod \"myzep-79945ccb7f-rz8ld_myzep(4ae2623a-8519-11e8-9f37-005056021169)\" failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod \"myzep-79945ccb7f-rz8ld_myzep\" network: failed to allocate for range 0: no IP addresses available in range set: 10.42.0.1-10.42.0.254"
I0711 15:11:18.026816   24727 kubelet.go:1906] SyncLoop (PLEG): "myzep-79945ccb7f-rz8ld_myzep(4ae2623a-8519-11e8-9f37-005056021169)", event: &pleg.PodLifecycleEvent{ID:"4ae2623a-8519-11e8-9f37-005056021169", Type:"ContainerDied", Data:"2a005a4267ec138567f2f17fe02adf30bbaf1c6e7eb043c6f0d48eda1e290ec4"}
W0711 15:11:18.026892   24727 pod_container_deletor.go:77] Container "2a005a4267ec138567f2f17fe02adf30bbaf1c6e7eb043c6f0d48eda1e290ec4" not found in pod's containers
I0711 15:11:18.026915   24727 kubelet.go:1906] SyncLoop (PLEG): "myzep-79945ccb7f-rz8ld_myzep(4ae2623a-8519-11e8-9f37-005056021169)", event: &pleg.PodLifecycleEvent{ID:"4ae2623a-8519-11e8-9f37-005056021169", Type:"ContainerStarted", Data:"956daf15f93582bfb4afbeb2b403cbd87da594860677b7ccd64cd22b80891484"}
I0711 15:11:18.065836   24727 kubelet.go:1906] SyncLoop (PLEG): "kube-dns-5ccb66df65-hmq9g_kube-system(a3d09f02-844e-11e8-acc8-005056021169)", event: &pleg.PodLifecycleEvent{ID:"a3d09f02-844e-11e8-acc8-005056021169", Type:"ContainerDied", Data:"ba02d61b270dbd8c8ade7112ac438f5bb3cdc007814e2342835da6fefca137c1"}
W0711 15:11:18.065904   24727 pod_container_deletor.go:77] Container "ba02d61b270dbd8c8ade7112ac438f5bb3cdc007814e2342835da6fefca137c1" not found in pod's containers
I0711 15:11:18.107722   24727 kubelet.go:1906] SyncLoop (PLEG): "cattle-cluster-agent-786699d558-qbs5d_cattle-system(e66e2af6-8516-11e8-9f37-005056021169)", event: &pleg.PodLifecycleEvent{ID:"e66e2af6-8516-11e8-9f37-005056021169", Type:"ContainerDied", Data:"c41354475b3889ae8097d1281bdde90ac12302cfa7a8509973f1edafe1816788"}
W0711 15:11:18.107784   24727 pod_container_deletor.go:77] Container "c41354475b3889ae8097d1281bdde90ac12302cfa7a8509973f1edafe1816788" not found in pod's containers
I0711 15:11:18.107805   24727 kubelet.go:1906] SyncLoop (PLEG): "cattle-cluster-agent-786699d558-qbs5d_cattle-system(e66e2af6-8516-11e8-9f37-005056021169)", event: &pleg.PodLifecycleEvent{ID:"e66e2af6-8516-11e8-9f37-005056021169", Type:"ContainerDied", Data:"e58444ae7afeca6d2bdbc05f58977adfbfccc2f4763fe09d718f895b74e9689c"}
W0711 15:11:18.107850   24727 pod_container_deletor.go:77] Container "e58444ae7afeca6d2bdbc05f58977adfbfccc2f4763fe09d718f895b74e9689c" not found in pod's containers
I0711 15:11:18.145772   24727 kubelet.go:1906] SyncLoop (PLEG): "kube-dns-autoscaler-6c4b786f5-jmglw_kube-system(a3c6720c-844e-11e8-acc8-005056021169)", event: &pleg.PodLifecycleEvent{ID:"a3c6720c-844e-11e8-acc8-005056021169", Type:"ContainerDied", Data:"61c15150c2862278ad4fc06c9647b0f8c7498e55602fef2a4d3e418087ba5e71"}
W0711 15:11:18.145834   24727 pod_container_deletor.go:77] Container "61c15150c2862278ad4fc06c9647b0f8c7498e55602fef2a4d3e418087ba5e71" not found in pod's containers
I0711 15:11:18.145856   24727 kubelet.go:1906] SyncLoop (PLEG): "kube-dns-autoscaler-6c4b786f5-jmglw_kube-system(a3c6720c-844e-11e8-acc8-005056021169)", event: &pleg.PodLifecycleEvent{ID:"a3c6720c-844e-11e8-acc8-005056021169", Type:"ContainerStarted", Data:"d691a819b39e8608f492f37bbbf237dcd525cb0ee2652038ee38c9e20c2f7277"}
I0711 15:11:18.365963   24727 kuberuntime_manager.go:403] No ready sandbox for pod "kube-dns-5ccb66df65-hmq9g_kube-system(a3d09f02-844e-11e8-acc8-005056021169)" can be found. Need to start a new one
I0711 15:11:18.407849   24727 kuberuntime_manager.go:403] No ready sandbox for pod "cattle-cluster-agent-786699d558-qbs5d_cattle-system(e66e2af6-8516-11e8-9f37-005056021169)" can be found. Need to start a new one
I0711 15:11:18.448430   24727 kuberuntime_manager.go:403] No ready sandbox for pod "kube-dns-autoscaler-6c4b786f5-jmglw_kube-system(a3c6720c-844e-11e8-acc8-005056021169)" can be found. Need to start a new one
E0711 15:11:18.864495   24727 cni.go:259] Error adding network: failed to allocate for range 0: no IP addresses available in range set: 10.42.0.1-10.42.0.254
E0711 15:11:18.864532   24727 cni.go:227] Error while adding to cni network: failed to allocate for range 0: no IP addresses available in range set: 10.42.0.1-10.42.0.254
E0711 15:11:18.917658   24727 cni.go:259] Error adding network: failed to allocate for range 0: no IP addresses available in range set: 10.42.0.1-10.42.0.254
E0711 15:11:18.917687   24727 cni.go:227] Error while adding to cni network: failed to allocate for range 0: no IP addresses available in range set: 10.42.0.1-10.42.0.254
E0711 15:11:18.935602   24727 cni.go:259] Error adding network: failed to allocate for range 0: no IP addresses available in range set: 10.42.0.1-10.42.0.254
E0711 15:11:18.935629   24727 cni.go:227] Error while adding to cni network: failed to allocate for range 0: no IP addresses available in range set: 10.42.0.1-10.42.0.254
E0711 15:11:19.052114   24727 remote_runtime.go:92] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "kube-dns-5ccb66df65-hmq9g_kube-system" network: failed to allocate for range 0: no IP addresses available in range set: 10.42.0.1-10.42.0.254
E0711 15:11:19.052164   24727 kuberuntime_sandbox.go:54] CreatePodSandbox for pod "kube-dns-5ccb66df65-hmq9g_kube-system(a3d09f02-844e-11e8-acc8-005056021169)" failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "kube-dns-5ccb66df65-hmq9g_kube-system" network: failed to allocate for range 0: no IP addresses available in range set: 10.42.0.1-10.42.0.254
E0711 15:11:19.052182   24727 kuberuntime_manager.go:646] createPodSandbox for pod "kube-dns-5ccb66df65-hmq9g_kube-system(a3d09f02-844e-11e8-acc8-005056021169)" failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "kube-dns-5ccb66df65-hmq9g_kube-system" network: failed to allocate for range 0: no IP addresses available in range set: 10.42.0.1-10.42.0.254

**Comment by gioppoluca** [comment]
I followed some hints on this issue:
https://github.com/cloudnativelabs/kube-router/issues/383
and solved I executed:
rm -rf /var/lib/cni/networks/cbr0/*
and restarted the host, now there are available IP for the cluster.

How I got there ending the IP? Probably because I'm using this host to test rancher and had to cleanup rancher a lot of time removing all the containers and creating new clusters a lot of times so this could had left no IP available since probably or I or the system did not clean the IP well.
Please either document the proper way to clean a cluster (it seems that just deleting it from the UI does not clean all.

**Comment by superseb** [comment]
Thanks for filing the issue, this should be covered in https://github.com/rancher/docs/pull/430

**Comment by Teskuroi** [comment]
I was facing the issue using RKE with Nodes on Rancher OS. That was happened after hard reboot on host machine.

Was getting from Pods:
`
Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "784b79139d521e83cad45de820efaddbfd83a6d3123e820ef572c36927cfcccb" network for pod "contactus-9df5bccf-f4vww": NetworkPlugin cni failed to set up pod "contactus-9df5bccf-f4vww_default" network: failed to allocate for range 0: no IP addresses available in range set: 10.42.8.1-10.42.8.254
`

Steps helped me:
1. Drain Node.
2. Connect using SSH to Node. And log in as `root`.
3. Check if `/opt/rke/var/lib/cni/networks` exists. If not, the instruction won't work for you.
3. Stop Docker service:
    ```bash
    ros service stop docker
    ```
4. Backup and recreate `network` directory:
    ```bash
    mv /opt/rke/var/lib/cni/networks /opt/rke/var/lib/cni/networks.backup
    mkdir /opt/rke/var/lib/cni/networks
    ```
5. Start Docker service:
    ```bash
    ros service start docker
    ```
6. Uncordon Node.

Hope that would help somebody else.
